docarray/array/storage/base/seqlike.py

```diff
@@ -10,6 +10,13 @@ class BaseSequenceLikeMixin(MutableSequence[Document]):
     def _update_subindices_append_extend(self, value):
         if getattr(self, '_subindices', None):
             for selector, da in self._subindices.items():
+
+                if getattr(da, '_config', None):
+                    if da._config.root_id:
+                        for i in range(len(value)):
+                            for doc in DocumentArray(value[i])[selector]:
+                                doc.tags['root_id'] = value[i].id
```

Comment:
I thank here you can use a built-in notation, something like
```python
subindex_docs = DocumentArray(value[i])[selector]
subindex_docs[:, 'tags__root_id'] = [value[i].id for _ in range(len(subindex_docs))]
```
Not 100% sure, but I think something like that would work so you don't have to loop manually

Reply:
This cannot work because here https://github.com/docarray/docarray/blob/main/docarray/array/storage/base/getsetdel.py#L301